### PR TITLE
Fix: Allow overrides on content objects (fixes #12)

### LINKED ISSUE #12: Option to make logo Menu or page only
An option to make the logo appear only in the nav bar when the learner is viewing the menu or page would be useful. Currently it will only appear in both without custom css.

### PR DESCRIPTION
**Pull request by swashbuck**
Fixes #12 

### Fix
* Adds `_isHidden` option to hide the logo on the main menu (in _course.json_)
* Content objects will inherit logo's `_src` and `alt` values from _course.json_
* Removes version number from README.md

### CODE REVIEW

**Comment by swashbuck** [comment]
> I'm finding that the _contentObject_ config is preventing the _course_ config being applied globally.

Thanks, Kirsty! This will need a bit of a refactor then. The AAT always complicates things 😛 

@kirsty-hames @StuartNicholls Do we need the ability to override the logo image for specific pages? Or are we fine with a single logo image that's always defined at the course level? Then, it can be shown or hidden on specific pages and the menu.

**Comment by swashbuck** [comment]
@kirsty-hames @StuartNicholls I've added a new `_isHidden` property (only to the documentation, example.json, and schemas). Let me know if this will work, and I'll implement it in the code.

**Comment by swashbuck** [comment]
@kirsty-hames @StuartNicholls I went ahead with the `_isHidden` option for _course.json_. PR description is updated above.

**Comment by kirsty-hames** [comment]
> a single logo image that's always defined at the course level? Then, it can be shown or hidden on specific pages and the menu.

Apologies for the delayed response but I agree with your suggestion, a single logo image with the option show/hide for specific pages and menu.

**Comment by kirsty-hames** [comment]
> * Changes `src` to `_src` to align with `_mobileSrc`

Looking at image source defined elsewhere in Adapt, `src` does seem to be the majority. e.g. `_graphic.src` for [Boxmenu items and components](https://github.com/adaptlearning/adapt_framework/blob/09b85b0a1eec18236c13f1310a09b9d171ee73b1/src/course/en/components.json#L354). Should we be using `src` and `mobileSrc` for consistency?

Note, [boxmenu logo graphic](https://github.com/adaptlearning/adapt-contrib-boxMenu/blob/df22de08aee5ab3395de44fd4a1e9491bd0be888/example.json#L5) does use `_src` but this was the only non `src` instance I could find.

**Comment by StuartNicholls** [comment]
> > * Changes `src` to `_src` to align with `_mobileSrc`
> 
> Looking at image source defined elsewhere in Adapt, `src` does seem to be the majority. e.g. `_graphic.src` for [Boxmenu items and components](https://github.com/adaptlearning/adapt_framework/blob/09b85b0a1eec18236c13f1310a09b9d171ee73b1/src/course/en/components.json#L354). Should we be using `src` and `mobileSrc` for consistency?
> 
> Note, [boxmenu logo graphic](https://github.com/adaptlearning/adapt-contrib-boxMenu/blob/df22de08aee5ab3395de44fd4a1e9491bd0be888/example.json#L5) does use `_src` but this was the only non `src` instance I could find.

Should this not be _small and _large so it reflects the breakpoints?

**Comment by StuartNicholls** [comment]
We could do with improving/rethinking the CSS in my humble opinion (or maybe I'm missing something) but perhaps thats a separate issue anyways.

**Comment by oliverfoster** [comment]
The _ is short for not translatable. All resource URLs should start with _. src without the underscore is a mistake that was replicated.

**Comment by kirsty-hames** [comment]
> The _ is short for not translatable. All resource URLs should start with _. src without the underscore is a mistake that was replicated.

Just so I'm clear @oliverfoster, are we saying to use `_src` moving forward and it's a known issue that we have `src` used elsewhere?

**Comment by StuartNicholls** [comment]
@kirsty-hames I think that's just an issue here, but yep, the underscore indicates 'do not translate' etc. My point is that elsewhere we use the breakpoint terminology, fixing this will be a breaking change no? Might as well update?

https://github.com/adaptlearning/adapt-contrib-graphic/blob/master/example.json

**Comment by kirsty-hames** [comment]
> Should this not be _small and _large so it reflects the breakpoints?

I'd agree, it does seem we use `src` for a single image source and `large`/`small` for responsive image sources in Adapt (noting these should be prefixed by _). If this is something that needs discussion perhaps it's best to leave the breaking change out of this PR and raise as a separate issue? And keep this PR for the content object overrides only.

**Comment by kirsty-hames** [comment]
> @kirsty-hames @StuartNicholls I went ahead with the `_isHidden` option for _course.json_. PR description is updated above. This should work with the AAT now (tested in legacy AAT).

This is working as expected thanks @swashbuck.

**Comment by oliverfoster** [comment]
> > The _ is short for not translatable. All resource URLs should start with _. src without the underscore is a mistake that was replicated.
> 
> Just so I'm clear @oliverfoster, are we saying to use `_src` moving forward and it's a known issue that we have `src` used elsewhere?

Yeas

**Comment by oliverfoster** [comment]
> @kirsty-hames I think that's just an issue here, but yep, the underscore indicates 'do not translate' etc. My point is that elsewhere we use the breakpoint terminology, fixing this will be a breaking change no? Might as well update?
> 
> https://github.com/adaptlearning/adapt-contrib-graphic/blob/master/example.json

Yes, which is why it hasn't yet been fixed in the graphic schema / example. It is however fixed in the jsx template.
https://github.com/adaptlearning/adapt-contrib-core/blob/b20e6f85edd597007e2846152d64996e07f1ee10/templates/image.jsx#L12-L21

**Comment by StuartNicholls** [comment]
@oliverfoster Completely missed there wasn't an underscore there!

**Comment by swashbuck** [comment]
Ok, I've reverted the `src` property change and will raise a new PR. Otherwise, let me know if you need any other changes.

**Review by kirsty-hames — APPROVED** [review]

**Comment by swashbuck** [comment]
Ok to merge, @StuartNicholls ?

**Comment by swashbuck** [comment]
> I think we want the logo to appear on content object pages if there is no config present for framework builds.

@StuartNicholls Ok, I've made this change in [d6cbc15](https://github.com/cgkineo/adapt-navigationLogo/pull/15/commits/d6cbc15b0d463cd4a52fcebbacdc95aab5237101). Content objects no longer require a `_navigationLogo` property and will be enabled by default if the config is in _course.json_.

**Comment by swashbuck** [comment]
> Also, the default nav order is 1, which is possibly correct, but the default spacer order in framework is 0, so the logo appears on the right hand side. 

@StuartNicholls Will you file a new ticket for this one? It may be part of a larger discussion about nav order default values. Thanks!

**Review by StuartNicholls — APPROVED** [review]

**Comment by kirsty-hames** [comment]
> > Also, the default nav order is 1, which is possibly correct, but the default spacer order in framework is 0, so the logo appears on the right hand side.
> 
> @StuartNicholls Will you file a new ticket for this one? It may be part of a larger discussion about nav order default values. Thanks!

Setting the nav order to 0 does mean the logo displays to the left of the spacer but agree this would be better raised as a core issue to review the nav order as a collection. On projects I typically use increments of 10 (minimum) as it gives a bit of flexibility when appending new nav items. I think this approach would be useful for setting defaults as it allows wriggle room for non-core plugins to fit around the core elements.

**Comment by kirsty-hames** [comment]
> > I think we want the logo to appear on content object pages if there is no config present for framework builds.
> 
> @StuartNicholls Ok, I've made this change in [d6cbc15](https://github.com/cgkineo/adapt-navigationLogo/pull/15/commits/d6cbc15b0d463cd4a52fcebbacdc95aab5237101). Content objects no longer require a `_navigationLogo` property and will be enabled by default if the config is in _course.json_.

Can confirm this is working as described. Testing on Mac Safari, FF, Chrome and iPhone

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 ## Settings overview
 
-The image displays with minimal padding by default or can be configured to fill the navigation bar height. For mobile, an alternative, mobile-friendly image can be specified or the logo can be hidden entirely.
+The image displays with minimal padding by default or can be configured to fill the navigation bar height. For mobile, an alternative, mobile-friendly image can be specified or the logo can be hidden entirely. The logo can be shown on the menu, on specific pages, or everywhere.
```
Comment (kirsty-hames) [inline README.md:9]: Perhaps add the default display to the last line.

e.g. The logo can be shown on the menu, on specific pages, or everywhere as default.

**Review by kirsty-hames — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (swashbuck) [reply README.md]: Done in e62e714

**Review by kirsty-hames — APPROVED** [review]
Working as expected thanks @swashbuck. Re-tested on Mac Safari, FF, Chrome and iPhone.

**Comment by guywillis** [comment]
Just a note to say the `_isHidden` property should probably be called something a bit more explicit to imply what the config option does. At present, `_isEnabled` and `_isHidden` both imply similar functionality whilst something like `_isHiddenOnMenu` is more descriptive.

**Comment by swashbuck** [comment]
> Just a note to say the `_isHidden` property should probably be called something a bit more explicit to imply what the config option does. At present, `_isEnabled` and `_isHidden` both imply similar functionality whilst something like `_isHiddenOnMenu` is more descriptive.

Thanks, @guywillis . Will you mind raising a new ticket for this? Alternatively, we have a ticket ( #16  ) to align image source properties, so we could tack it on to that?

**Comment by kirsty-hames** [comment]
> > Just a note to say the `_isHidden` property should probably be called something a bit more explicit to imply what the config option does. At present, `_isEnabled` and `_isHidden` both imply similar functionality whilst something like `_isHiddenOnMenu` is more descriptive.
> 
> Thanks, @guywillis . Will you mind raising a new ticket for this? Alternatively, we have a ticket ( #16 ) to align image source properties, so we could tack it on to that?

Both the readme text and the schema title note this hides the logo on the menu.

**Comment by guywillis** [comment]
> Both the readme text and the schema title note this hides the logo on the menu.

It's great to have the functionality defined in documentation but I do think the naming of config options should try to succinctly describe functionality too.
 
> Thanks, @guywillis . Will you mind raising a new ticket for this? Alternatively, we have a ticket ( #16 ) to align image source properties, so we could tack it on to that?

Sure, I'll raise a new ticket.

**Pull request merged**